database/user.go

```diff
@@ -54,19 +56,20 @@ type User struct {
 	db  *Database
 	log log.Logger
 
-	MXID           id.UserID
-	DiscordID      string
-	DiscordToken   string
-	ManagementRoom id.RoomID
-	SpaceRoom      id.RoomID
-	DMSpaceRoom    id.RoomID
+	MXID             id.UserID
+	DiscordID        string
+	DiscordToken     string
+	ManagementRoom   id.RoomID
+	SpaceRoom        id.RoomID
+	DMSpaceRoom      id.RoomID
+	HeartbeatSession *discordgo.HeartbeatSession
 
 	ReadStateVersion int
 }
 
 func (u *User) Scan(row dbutil.Scannable) *User {
-	var discordID, managementRoom, spaceRoom, dmSpaceRoom, discordToken sql.NullString
-	err := row.Scan(&u.MXID, &discordID, &discordToken, &managementRoom, &spaceRoom, &dmSpaceRoom, &u.ReadStateVersion)
+	var discordID, managementRoom, spaceRoom, dmSpaceRoom, discordToken, heartbeatSessionJSON sql.NullString
+	err := row.Scan(&u.MXID, &discordID, &discordToken, &managementRoom, &spaceRoom, &dmSpaceRoom, &u.ReadStateVersion, &heartbeatSessionJSON)
```

Comment:
```suggestion
	var discordID, managementRoom, spaceRoom, dmSpaceRoom, discordToken sql.NullString
	err := row.Scan(&u.MXID, &discordID, &discordToken, &managementRoom, &spaceRoom, &dmSpaceRoom, &u.ReadStateVersion, dbutil.JSON{Data: &u.HeartbeatSession})
```

Reply:
Looks like this didn't get applied, there shouldn't be any need to unmarshal manually since dbutil.JSON supports reading from the db (it'll ignore nulls too)

Reply:
oops, missed that sorry! and woah that's so neat